pkg/appfile/addon.go

```diff
@@ -0,0 +1,199 @@
+			if err != nil {
+				return nil, err
+			}
+			if err := os.Chdir(cwd); err != nil {
+				return nil, err
+			}
+
+			outputList := strings.Split(strings.ReplaceAll(string(outputs), " ", ""), "\n")
+			if outputList[len(outputList)-1] == "" {
+				outputList = outputList[:len(outputList)-1]
+			}
+			if err := generateSecretFromTerraformOutput(k8sClient, outputList, name, namespace); err != nil {
+				return nil, err
+			}
+		}
+	}
+	return nativeVelaComponents, nil
+}
+
+func callTerraform(tfJSONDir string) ([]byte, error) {
+	if err := os.Chdir(tfJSONDir); err != nil {
+		return nil, err
+	}
+	var cmd *exec.Cmd
+	cmd = exec.Command("bash", "-c", "terraform init")
+	if err := common.RealtimePrintCommandOutput(cmd, TerraformLog); err != nil {
+		return nil, err
+	}
+
+	cmd = exec.Command("bash", "-c", "terraform apply --auto-approve")
+	if err := common.RealtimePrintCommandOutput(cmd, TerraformLog); err != nil {
+		return nil, err
+	}
+
+	// Get output from Terraform
+	cmd = exec.Command("bash", "-c", "terraform output")
+	outputs, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+	return outputs, nil
+}
+
+// generateSecretFromTerraformOutput generates secret from Terraform output
+func generateSecretFromTerraformOutput(k8sClient client.Client, outputList []string, name, namespace string) error {
+	ctx := context.TODO()
+	err := k8sClient.Create(ctx, &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
+	if err == nil {
+		return fmt.Errorf("namespace %s doesn't exist", namespace)
+	}
+	var cmData = make(map[string]string, len(outputList))
+	for _, i := range outputList {
+		line := strings.Split(i, "=")
+		if len(line) != 2 {
+			return fmt.Errorf("Terraform output isn't in the right format")
+		}
+		k := strings.TrimSpace(line[0])
+		v := strings.TrimSpace(line[1])
+		if k != "" && v != "" {
+			cmData[k] = v
+		}
+	}
+
+	objectKey := client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}
+	var secret v1.Secret
+	if err := k8sClient.Get(ctx, objectKey, &secret); err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("retrieving the secret from cloud resource %s hit an issue: %w", name, err)
+	} else if err == nil {
+		if err := k8sClient.Delete(ctx, &secret); err != nil {
+			return fmt.Errorf("failed to store cloud resource %s output to secret: %w", name, err)
+		}
+	}
+
+	secret = v1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		StringData: cmData,
+	}
+
+	if err := k8sClient.Create(ctx, &secret); err != nil {
+		return fmt.Errorf("failed to store cloud resource %s output to secret: %w", name, err)
+	}
+	return nil
+}
+
+// getTerraformJSONFiles gets Terraform JSON files or modules from workload
+func getTerraformJSONFiles(k8sClient client.Client, wl *Workload, applicationName string, namespace string) ([]byte, error) {
+	pCtx := process.NewContext(wl.Name)
```

Comment:
This function has too many boilerplate code that doesn't have anything to do with Terraform data.
Would you refactor the code to reduce duplication and make the code cleaner?

Reply:
Done, thanks.